src/EIO.hs

```diff
@@ -62,9 +63,10 @@ safeMain = EIO.do
 
 @since 0.0.0.0
 -}
-runEIO :: EIO '[] () -> IO ()
+runEIO :: DisallowUnhandledExceptions excepts => EIO excepts () -> IO ()
```

Comment:
Being new to custom type errors, and type level programming in general, I wasn't sure if there was a cleaner way to convey to the consumer of this function that excepts is constrained to the kind `[Type]`